pytrnsys/trnsys_util/buildTrnsysDeck.py

```diff
@@ -348,4 +343,42 @@ def replaceLines(self, replaceList):
             for index, line in enumerate(self.linesChanged):
                 if oldLine in line:
                     self.linesChanged[index] = newLine + "\n"
-                    changedLine = oldLine
+
+    def analyseDck(self):
+        maxLineWidth = 0
+        maxNumberOfConstantsInABlock = 0
+
+        constantsToCheck = ["UNIT", "EQUATIONS", "CONSTANTS", "PARAMETERS", "INPUTS"]
+        numOfTrnsysConstants = {}
+
+        for index, line in enumerate(self.deckText):
+            maxLineWidth = max(maxLineWidth, len(line))
+            if maxLineWidth > 1000:
+                self.logger.warning(f"Line {index + 1} has {maxLineWidth} characters which exceeds the limit.")
+                maxLineWidth = 0
+
+            for constant in constantsToCheck:
+                match = _re.search(fr"^\b{constant}\s*\d+\b", line, _re.MULTILINE)
+                if match:
+                    if constant == "UNIT":
+                        numOfTrnsysConstants[constant] = numOfTrnsysConstants.get(constant, 0) + 1
+                    elif constant == "CONSTANTS":
+                        split = match.group().split()
+                        numOfTrnsysConstants["EQUATIONS"] = numOfTrnsysConstants.get("EQUATIONS", 0) + int(split[1])
```

Comment:
"CONSTANTS" vs "EQUATIONS"...isn't there a mismatch?

Reply:
> "CONSTANTS" vs "EQUATIONS"...isn't there a mismatch?

If I am not mistaken, "CONSTANTS" is considered as "EQUATIONS" as per discussion with @martin-neugebauer.

Reply:
Indeed the TrnsysConstant `nMaxEquations` specifies the maximum for the sum of `EQUATIONS` and `CONSTANTS`, so this approach is correct.

Reply:
ok, but shouldn't it be something like `elif constant in ("CONSTANTS", "EQUATIONS")` then?

Reply:
I'm lacking a bit the code overview in this particular location, but both `EQUATIONS` as well as `CONSTANTS` should be added to the same counter. So if `EQUATIONS` is not added somewhere else, then I agree with @zuckerruebe. One other thing: Why is `UNIT` added to `constant`?